lib/internal/http2/core.js

```diff
@@ -2214,6 +2176,30 @@ class Http2Stream extends Duplex {
   }
 }
 
+function onTimeout(kSession) {
+  // If the session is destroyed, this should never actually be invoked,
+  // but just in case...
+  if (this.destroyed)
+    return;
+  // This checks whether a write is currently in progress and also whether
+  // that write is actually sending data across the write. The kHandle
+  // stored `chunksSentSinceLastWrite` is only updated when a timeout event
+  // happens, meaning that if a write is ongoing it should never equal the
+  // newly fetched, updated value.
+  if (this[kState].writeQueueSize > 0) {
+    const handle = kSession ? this[kSession][kHandle] : this[kHandle];
+    const chunksSentSinceLastWrite = handle !== undefined ?
```

Comment:
Nit: this code can probably be nicer with nullish coalescing.

Reply:
Thanks for the suggestion, but I'm reluctant to use the `nullish coalescing` syntax because it can't be backported to an older release.